rosbag2_transport/test/rosbag2_transport/test_record_all.cpp

```diff
@@ -60,17 +61,22 @@ TEST_F(RecordIntegrationTestFixture, published_messages_from_multiple_topics_are
   ASSERT_TRUE(pub_manager.wait_for_matched(array_topic.c_str()));
   ASSERT_TRUE(pub_manager.wait_for_matched(string_topic.c_str()));
 
+  // At this point, we expect that the topics /string_topic, /array_topic, and /events/write_split
+  // are available to be recorded.  However, wait_for_matched() only checks for /string_topic
+  // and /array_topic, so ask the recorder to make sure it has successfully subscribed to all.
+  ASSERT_TRUE(recorder->wait_for_topic_to_be_discovered("/events/write_split"));
+
```

Comment:
@clalancette What a point to wait for the `"/events/write_split"` on rosbag2_recorder side?
The `"/events/write_split"` is an internal `rosbag2_recorder` service event that we are not going to use in this test scenario.

Reply:
Yeah, I guess that's true.  I'll remove this one.